src/service/resource.js

```diff
@@ -103,6 +104,15 @@ export class Resource {
       Resource.forElementOptional(element).updateOwner(owner);
     }
     element[OWNER_PROP_] = owner;
+    // Need to clear owner cache for all child elements
```

Comment:
I switched to `getElementByClassName('-amp-element')` which finds me all the amp children. 
It's same concept as `discoverResourcesForElement_` except `discoverResourcesForElement` doesn't work in case `<owner><child><grandchild><greatgrandchild>` because it will not clear cache for greatgrandchild. 
PTAL